### PR TITLE
Add `wp i18n` command to generate POT files

### PR DESCRIPTION
**Pull request by fmfernandes**
#### Changes proposed in this Pull Request

* This PR adds the `i18n` command from [WP-CLI](https://github.com/wp-cli/i18n-command) to generate POT files.

#### Testing instructions

* Run `composer update`.
* There's a composer script `makepot` that should help as an alias to the `i18n` command. To build POT for the theme `build-processes-demo` we could run it like so:
  * `composer makepot:theme`.
  * Note that there's no `__` function anywhere in the theme, so you must first add one for it to be included in the POT file.
* We can also generate POT files for the `mu-plugin/build-processes-demo-blocks`:
  * `composer makepot:blocks`.

#### Notes

Since I was working on adding `phpcs` as a composer dependency as well I thought that we might use `wp i18n` for POT generation instead of a npm package. We can even remove the composer dependency if we assume everyone has WP-CLI installed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,6 +25,7 @@ themes/*
 # Other wp-content exclusions
 upgrade/*
 uploads/*
+languages/*
```
[the remark this answers is not in the document]
Comment (fmfernandes) [reply .gitignore:28]: Dealing with POT files made me realize we'll need it.

**Comment by fmfernandes** [comment]
> Do you have an example of how/when that would happen? I'm just trying to understand because I can't think of an example, but it looks like you've stumbled upon one!

I have a folder structure like this: `~/GitHub/plugins/`. If you clone this repo inside it, and run `composer php:lint` (or `php:fix`) PHPCS will just register the sniffs and not actually process the files. The same will happen if the repo is cloned inside a `node_modules`/`vendor`/`bin` folder but that seem way less likely than a `plugins/` folder. But I can add a `ˆ` to the other ones as well if needed 😅 

I initially added a fix for that in this [PR](https://github.com/a8cteam51/build-processes-demo/pull/12#issue-1469759298) (in the notes section), slightly different way but I guess both works.

> For https://github.com/a8cteam51/build-processes-demo/issues/15, can you also include an example of how to register the translations for the mu-plugins/build-processes-demo-extra-functionality plugin? Thank you!

Ops! For sure!

**Comment by fmfernandes** [comment]
Hey @ahegyes, just added some translations for the `build-processes-demo-extra-functionality` plugin.

**Comment by ahegyes** [comment]
Great work on this @fmfernandes! I mostly made a few changes in `composer.json` for the workflow of editing translations, and then I merged the branch already to test it live. 

Translations from all 4 sources are loading nicely:

**Comment by ahegyes** [comment]
For posterity, I changed all the textdomain-loading actions to `init` for compatibility with plugins like WPML and Polylang. Reference: https://developer.wordpress.org/reference/functions/load_plugin_textdomain/#comment-1568

**Pull request merged**